internal/runners/clean/run_win.go

```diff
@@ -4,34 +4,57 @@ package clean
 
 import (
 	"fmt"
-	"os/exec"
+	"os"
+	"path/filepath"
 
 	"github.com/gobuffalo/packr"
 
+	"github.com/ActiveState/cli/internal/exeutils"
 	"github.com/ActiveState/cli/internal/language"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/scriptfile"
 )
 
-func removeConfig(cfg configurable) error {
-	return runScript("removeConfig", cfg.ConfigPath())
+func (u *Uninstall) runUninstall() error {
+	err := removeCache(u.cfg.CachePath())
+	if err != nil {
+		return err
+	}
+
+	err = removePaths(u.installPath, u.cfg.ConfigPath())
+	if err != nil {
+		return err
+	}
+
+	u.out.Print(locale.T("clean_success_message"))
+	return nil
 }
 
-func removeInstall(installPath string) error {
-	return runScript("removeInstall", installPath)
+func removeConfig(configPath string) error {
+	return removePaths(configPath)
 }
 
-func runScript(scriptName, path string) error {
+func removePaths(dirs ...string) error {
+	logging.Debug("Removing paths: %v", dirs)
+	scriptName := "removePaths"
 	box := packr.NewBox("../../../assets/scripts/")
 	scriptBlock := box.String(fmt.Sprintf("%s.bat", scriptName))
 	sf, err := scriptfile.New(language.Batch, scriptName, scriptBlock)
 	if err != nil {
-		return err
+		return locale.WrapError(err, "err_clean_script", "Could not create new scriptfile")
 	}
 
-	cmd := exec.Command("cmd.exe", "/C", sf.Filename(), path)
-	err = cmd.Start()
+	exe, err := os.Executable()
 	if err != nil {
-		return err
+		return locale.WrapError(err, "err_clean_executable", "Could not get executable name")
+	}
+
+	args := []string{"/C", sf.Filename(), fmt.Sprintf("%d", os.Getpid()), filepath.Base(exe)}
+	args = append(args, dirs...)
+	_, err = exeutils.ExecuteAndForget("cmd.exe", args...)
```

Comment:
This doesn't feel like it should be using `..AndForget()`. When `state clean` finishes I'd expect it to actually be done cleaning. If that's not possible for some reason we need to ensure that the UX properly conveys this.

Reply:
On Windows, we can't remove anything until the `state.exe` process has ended. I've updated the messaging for `state clean uninstall` on Windows though I'm not sure if that's what you had in mind.

Reply:
I am thinking about the optimal un-installation experience for Windows:

First of all I believe that we should have a roll-back mechanism in case anything during the un-installation fails.  That is true for all platforms and I created a ticket for that: https://www.pivotaltracker.com/story/show/178134918  But that is something different. :)

- On Windows we cannot remove the programme that we are currently running, so maybe we need to have some functionality, to know as well as possible that a script running in the background can actually remove the executable.  Maybe by checking that there are no other processes running the executable, and by moving it to a temporary directory (which should ensure that we have permissions to delete it)?
- Ideally, we would have an entry in the "Add/Remove application" in the Systems dialog...  Then, the OS would deal with all that crap.  But, of course, that means that we would have to publish entries to the registry with an installation database during the installation which is imitating MSI functionality...